### PR TITLE
Restructure the test suite to follow the code

### PR DESCRIPTION
**Pull request by Armavica**
**What is this PR about?**
This PR splits up several test files as discussed in #5777.

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...

## Docs / Maintenance
- Split up several test files to follow the code file hierarchy.

### CODE REVIEW

**Comment by Armavica** [comment]
I didn't touch yet: `test_posteriors`, `test_step`, `test_variational_inference`, but will take care of them soon.

I looked at `test_missing`, `test_shared` and `test_types` and couldn't decide where to put them. Would you have an idea about these three?

File: pymc/tests/distributions/test_multivariate.py [hunk 1]
```diff
@@ -1874,3 +1875,28 @@ def test_car_rng_fn(sparse):
...
+    assert posdef(data) == 1
+
+
+def test_posdef_symmetric2():
+    data = np.array([[1.0, 2], [2, 1]], dtype=aesara.config.floatX)
+    assert posdef(data) == 0
+
+
+def test_posdef_symmetric3():
+    """The test return 0 if the matrix has 0 eigenvalue.
```
Comment (canyon289) [inline pymc/tests/distributions/test_multivariate.py]:
```suggestion
    """The test returns 0 if the matrix has 0 eigenvalue.
```

File: pymc/tests/distributions/test_multivariate.py [hunk 2]
```diff
@@ -1874,3 +1875,28 @@ def test_car_rng_fn(sparse):
         )
         f -= 1
     assert p > delta
+
+
+def test_posdef_symmetric1():
```
Comment (canyon289) [inline pymc/tests/distributions/test_multivariate.py]: I know youre just restructuring these so this isnt required, just commenting. Would be nice to paramterizing these

**Comment by ricardoV94** [comment]
`test_missing` corresponds to functionality implemented in `model` I think, so `test_model` might be a good place. 

`test_shared` corresponds to uses of `pm.Data`? If so, wherever that is implemented.

**Comment by Armavica** [comment]
> `test_shared` corresponds to uses of `pm.Data`? If so, wherever that is implemented.

I don't think that it uses anything from `data.py`. It is directly using `aesara.shared`.

**Comment by canyon289** [comment]
looked through this and it LGTM. coverage is showing more test coverage as well. we should merge before theres any conflicts that arise and follow up in prs imo

**Review by canyon289 — APPROVED** [review]

**Comment by canyon289** [comment]
> I didn't touch yet: `test_posteriors`, `test_step`, `test_variational_inference`, but will take care of them soon.
> 
> I looked at `test_missing`, `test_shared` and `test_types` and couldn't decide where to put them. Would you have an idea about these three?

Lets do this in another PR to save you rebase headache. This many files changed always make me nervous and rebasing across file changes is a real pain

**Comment by canyon289** [comment]
Thanks much @Armavica

**Pull request merged**